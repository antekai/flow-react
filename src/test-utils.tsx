import { RenderHookResult, render, renderHook } from "@testing-library/react";

interface ProvidersForTestsProps {
  children: JSX.Element;
}

export const ProvidersForTests = ({ children }: ProvidersForTestsProps) => {
  return <div>{children}</div>;
};

const customRender = (ui: JSX.Element, options?: Record<string, unknown>) =>
  render(ui, {
    wrapper: ({ children }) => (
      <ProvidersForTests {...options}>{children}</ProvidersForTests>
    ),
  });

export const customRenderHook = <P, R>(
  hook: (props: P) => R,
  options?: Record<string, unknown>
): RenderHookResult<any, any> =>
  renderHook(hook, {
    wrapper: ({ children }) => (
      <ProvidersForTests {...options}>{children}</ProvidersForTests>
    ),
  });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
export { customRenderHook as renderHook };
