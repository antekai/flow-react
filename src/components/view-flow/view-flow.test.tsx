import { ViewFlow } from ".";
import { render } from "../../test-utils";

describe("ViewFlow", () => {
  const { asFragment } = render(<ViewFlow />);

  it("should render successfully", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
