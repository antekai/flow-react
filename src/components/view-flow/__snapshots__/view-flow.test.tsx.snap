// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ViewFlow > should render successfully 1`] = `
<DocumentFragment>
  <div
    class="StyledViewFlow-jDENTs bCHcBa"
  >
    <div
      class="react-flow"
      data-testid="rf__wrapper"
      style="width: 100%; height: 100%; overflow: hidden; position: relative; z-index: 0;"
    >
      <div
        class="react-flow__renderer"
        style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
      >
        <div
          class="react-flow__pane"
          style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
        >
          <div
            class="react-flow__viewport react-flow__container"
            style="transform: translate(0px,0px) scale(1);"
          >
            <svg
              class="react-flow__edges react-flow__container"
              height="500"
              style="z-index: 0;"
              width="500"
            >
              <defs />
              <g />
            </svg>
            <div
              class="react-flow__edgelabel-renderer"
            />
            <div
              class="react-flow__nodes"
              style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
            >
              <div
                aria-describedby="react-flow__node-desc-1"
                class="react-flow__node react-flow__node-input nopan selectable"
                data-id="1"
                data-testid="rf__node-1"
                role="button"
                style="z-index: 0; transform: translate(250px,5px); pointer-events: all; visibility: hidden;"
                tabindex="0"
              >
                Node 1
                <div
                  class="react-flow__handle react-flow__handle-bottom nodrag nopan source connectable connectablestart connectableend connectionindicator"
                  data-handlepos="bottom"
                  data-id="1-null-source"
                  data-nodeid="1"
                />
              </div>
              <div
                aria-describedby="react-flow__node-desc-1"
                class="react-flow__node react-flow__node-default nopan selectable"
                data-id="2"
                data-testid="rf__node-2"
                role="button"
                style="z-index: 0; transform: translate(100px,100px); pointer-events: all; visibility: hidden;"
                tabindex="0"
              >
                <div
                  class="react-flow__handle react-flow__handle-top nodrag nopan target connectable connectablestart connectableend connectionindicator"
                  data-handlepos="top"
                  data-id="2-null-target"
                  data-nodeid="2"
                />
                Node 2
                <div
                  class="react-flow__handle react-flow__handle-bottom nodrag nopan source connectable connectablestart connectableend connectionindicator"
                  data-handlepos="bottom"
                  data-id="2-null-source"
                  data-nodeid="2"
                />
              </div>
              <div
                aria-describedby="react-flow__node-desc-1"
                class="react-flow__node react-flow__node-default nopan selectable"
                data-id="3"
                data-testid="rf__node-3"
                role="button"
                style="z-index: 0; transform: translate(400px,100px); pointer-events: all; visibility: hidden;"
                tabindex="0"
              >
                <div
                  class="react-flow__handle react-flow__handle-top nodrag nopan target connectable connectablestart connectableend connectionindicator"
                  data-handlepos="top"
                  data-id="3-null-target"
                  data-nodeid="3"
                />
                Node 3
                <div
                  class="react-flow__handle react-flow__handle-bottom nodrag nopan source connectable connectablestart connectableend connectionindicator"
                  data-handlepos="bottom"
                  data-id="3-null-source"
                  data-nodeid="3"
                />
              </div>
              <div
                aria-describedby="react-flow__node-desc-1"
                class="react-flow__node react-flow__node-default nopan selectable"
                data-id="4"
                data-testid="rf__node-4"
                role="button"
                style="z-index: 0; transform: translate(400px,200px); pointer-events: all; visibility: hidden;"
                tabindex="0"
              >
                <div
                  class="react-flow__handle react-flow__handle-top nodrag nopan target connectable connectablestart connectableend connectionindicator"
                  data-handlepos="top"
                  data-id="4-null-target"
                  data-nodeid="4"
                />
                Custom Node
                <div
                  class="react-flow__handle react-flow__handle-bottom nodrag nopan source connectable connectablestart connectableend connectionindicator"
                  data-handlepos="bottom"
                  data-id="4-null-source"
                  data-nodeid="4"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <svg
        class="react-flow__background"
        data-testid="rf__background"
        style="position: absolute; width: 100%; height: 100%; top: 0px; left: 0px;"
      >
        <pattern
          height="20"
          id="pattern-1undefined"
          patternTransform="translate(-0.5,-0.5)"
          patternUnits="userSpaceOnUse"
          width="20"
          x="0"
          y="0"
        >
          <circle
            cx="0.5"
            cy="0.5"
            fill="#91919a"
            r="0.5"
          />
        </pattern>
        <rect
          fill="url(#pattern-1undefined)"
          height="100%"
          width="100%"
          x="0"
          y="0"
        />
      </svg>
      <div
        class="react-flow__panel react-flow__attribution bottom right"
        data-message="Please only hide this attribution when you are subscribed to React Flow Pro: https://pro.reactflow.dev"
        style="pointer-events: all;"
      >
        <a
          aria-label="React Flow attribution"
          href="https://reactflow.dev"
          rel="noopener noreferrer"
          target="_blank"
        >
          React Flow
        </a>
      </div>
      <div
        id="react-flow__node-desc-1"
        style="display: none;"
      >
        Press enter or space to select a node.You can then use the arrow keys to move the node around. Press delete to remove it and escape to cancel. 
      </div>
      <div
        id="react-flow__edge-desc-1"
        style="display: none;"
      >
        Press enter or space to select an edge. You can then press delete to remove it or escape to cancel.
      </div>
      <div
        aria-atomic="true"
        aria-live="assertive"
        id="react-flow__aria-live-1"
        style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(100%);"
      />
    </div>
  </div>
</DocumentFragment>
`;
